streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -325,13 +321,7 @@ boolean maybeResumeSuspendedTask(final TaskId taskId,
                     // we need to catch migration exception internally since this function
                     // is triggered in the rebalance callback
                     log.info("Failed to resume {} {} since it got migrated to another thread already. " +
-                        "Closing it as zombie before triggering a new rebalance.", taskTypeName, task.id());
-                    final RuntimeException fatalException = closeZombieTask(task);
```

Comment:
The new diff contains two `closeZombieTask` usages, where one of them in AssignedStreamTasks has
```
 firstException.compareAndSet(null, closeZombieTask(task));
 prevActiveTasks.remove(id);
```
which makes me wonder whether standby tasks also have to update some internal state like `closeRevokedStandbyTasks`, which makes me wonder should we disgard this function, as using it alone is usually not good enough and may encourage bugs since it gives people impression it could do more than just closing a task.

Reply:
Good question, but `StandbyTask#close` explicitly clarifies it can never be a zombie. With the removal of most occurrences of `closeZombieTask` from `AssignedTasks` there's actually a good amount of further cleanup to do here, especially around the general `close`
That's actually specifically what I started doing before I realized the changes were getting out of hand for this simple bugfix PR.